source/slang/ir-insts.h

```diff
@@ -333,6 +353,13 @@ struct IRWitnessTable : IRGlobalValue
     RefPtr<GenericDecl> genericDecl;
     DeclRef<Decl> subTypeDeclRef, supTypeDeclRef;
     IRValueList<IRWitnessTableEntry> entries;
+    virtual void dispose() override
+    {
```

Comment:
Oh, it isn't just decorations. IR values are using `dispose` too. Now I'm really confused.

Reply:
I guess just calling destructors should be just fine. The code was once just using destructors, but I changed it to explicit `dispose` calls at some point in an attempt to address an issue, and this may not be the actual solution for that issue but I did not change it back.